Sources/SwiftDocCUtilities/Transformers/StaticHostableTransformer.swift

```diff
@@ -0,0 +1,169 @@
+        }
+        self.indexHTMLData = Data(indexHTML.utf8)
+    }
+    
+    /// Creates a static hostable version of the documention in the data folder of an archive pointed to by the `dataProvider`
+    /// - Parameters:
+    ///   - outputURL: The folder where the output will be placed
+    ///   - basePath: The path to be prefix to all href and src parameters in generated html
+    /// - Returns: An array if problems encounter during the archive.
+    func transform() throws {
+
+        let node = dataProvider.fileSystem
+        
+        // We should be starting at the data folder of a .doccarchive.
+        switch node {
+        case .directory(let dir):
+            try processDirectoryContents(directoryRoot: outputURL, relativeSubPath: "", directoryContents: dir.children)
+        case .file(_):
+            throw StaticHostableTransformerError.dataProviderDoesNotReferenceValidInput
+        }
+    }
+
+
+    /// Create a directory at the provided URL
+    ///
+    private func createDirectory(url: URL) throws {
+        if !fileManager.fileExists(atPath: url.path) {
+            try fileManager.createDirectory(at: url, withIntermediateDirectories: true, attributes: [:])
+        }
+    }
+
+    /// Processes the contents of a given directory
+    /// - Parameters:
+    ///   - root: The root output URL
+    ///   - directory: The relative path (to the root) of the directory for which then content will processed.
+    ///   - nodes: The directory contents
+    /// - Returns: An array of problems that may have occured during processing
+    private func processDirectoryContents(directoryRoot: URL, relativeSubPath: String, directoryContents: [FSNode]) throws {
+
+        for node in directoryContents {
+            switch node {
+            case .directory(let dir):
+                try processDirectory(directoryRoot: directoryRoot, currentDirectoryNode: dir, directorySubPath: relativeSubPath)
+            case .file(let file):
+                let outputURL = directoryRoot.appendingPathComponent(relativeSubPath)
+                try processFile(file: file, outputURL: outputURL)
+            }
+        }
+
+    }
+
+    /// Processes the  given directory
+    /// - Parameters:
+    ///   - root: The root output URL
+    ///   - dir: The FSNode that represents the directory
+    ///   - currentDirectory: The relative path (to the root) of the directory that will contain this directory
+    /// - Returns: An array of problems that may have occured during processing
+    private func processDirectory(directoryRoot: URL, currentDirectoryNode: FSNode.Directory, directorySubPath: String) throws {
+
+        // Create the path for the new directory
+        var newDirectory = directorySubPath
+        let newPathComponent = currentDirectoryNode.url.lastPathComponent
+        if !newDirectory.isEmpty {
+            newDirectory += "/"
+        }
+        newDirectory += newPathComponent
```

Comment:
Any sort of string concatenation of paths raises alarm bells for me. I'm not 100% clear on what this method is doing, but could it use URLComponents or appendingPathComponent?